aries_vcx/src/handlers/connection/connection.rs

```diff
@@ -804,15 +772,15 @@ impl<'de> Deserialize<'de> for Connection {
     }
 }
 
-impl Into<(SmConnectionState, PairwiseInfo, CloudAgentInfo, String)> for Connection {
-    fn into(self) -> (SmConnectionState, PairwiseInfo, CloudAgentInfo, String) {
-        (self.state_object(), self.pairwise_info().to_owned(), self.cloud_agent_info().to_owned(), self.source_id())
+impl Into<(SmConnectionState, PairwiseInfo, CloudAgentInfo, String, String)> for Connection {
+    fn into(self) -> (SmConnectionState, PairwiseInfo, CloudAgentInfo, String, String) {
+        (self.state_object(), self.pairwise_info().to_owned(), self.cloud_agent_info().to_owned(), self.source_id(), self.get_thread_id().unwrap()) // TODO: Use TryInto instead
```

Comment:
`get_thread_id()` always returns `Ok()` - why not to just remove VcxResult wrapper from the return value?

Reply:
However, currently, invitee's SM has `thread_id` still undefined on creation - we should eventually create invitee in invited state (replace Invited state with Initial state with invitation and therefore thread id defined).